src/cat.js

```diff
@@ -25,103 +26,103 @@ module.exports = (common, options) => {
     after(() => common.clean())
 
     before(() => Promise.all([
-      ipfs.add(fixtures.smallFile.data),
-      ipfs.add(fixtures.bigFile.data)
+      all(ipfs.add(fixtures.smallFile.data)),
+      all(ipfs.add(fixtures.bigFile.data))
     ]))
 
     it('should cat with a base58 string encoded multihash', async () => {
-      const data = await ipfs.cat(fixtures.smallFile.cid)
+      const data = await concat(ipfs.cat(fixtures.smallFile.cid))
       expect(data.toString()).to.contain('Plz add me!')
     })
 
     it('should cat with a Buffer multihash', async () => {
-      const cid = Buffer.from(bs58.decode(fixtures.smallFile.cid))
+      const cid = new CID(fixtures.smallFile.cid).multihash
 
-      const data = await ipfs.cat(cid)
+      const data = await concat(ipfs.cat(cid))
       expect(data.toString()).to.contain('Plz add me!')
     })
 
     it('should cat with a CID object', async () => {
       const cid = new CID(fixtures.smallFile.cid)
 
-      const data = await ipfs.cat(cid)
+      const data = await concat(ipfs.cat(cid))
       expect(data.toString()).to.contain('Plz add me!')
     })
 
     it('should cat a file added as CIDv0 with a CIDv1', async () => {
       const input = Buffer.from(`TEST${Date.now()}`)
 
-      const res = await ipfs.add(input, { cidVersion: 0 })
+      const res = await all(ipfs.add(input, { cidVersion: 0 }))
 
       const cidv0 = new CID(res[0].hash)
       expect(cidv0.version).to.equal(0)
 
       const cidv1 = cidv0.toV1()
 
-      const output = await ipfs.cat(cidv1)
-      expect(output).to.eql(input)
+      const output = await concat(ipfs.cat(cidv1))
+      expect(output.slice()).to.eql(input)
```

Comment:
Why we need a shallow copy here (`.slice()`)?

Reply:
concat returns a BufferList